src/metabase/models/database.clj

```diff
@@ -177,7 +177,13 @@
                       {:status-code     400
                        :existing-engine existing-engine
                        :new-engine      new-engine}))
-      (u/prog1 (handle-secrets-changes database)
+      (u/prog1  (-> database
+                    (cond->
+                      ;; If the engine doesn't support nested field columns, `json_unfolding` must be nil
```

Comment:
where do we need things to behave differently with `false` and `nil`?

Reply:
When it comes to details json_unfolding=nil is treated the same as json_unfolding=true. See the implementation for json-unfolding-default in this namespace.
https://github.com/metabase/metabase/blob/f2e4d8e50a7f34bbe8e18a32cb967fa257d63eff/src/metabase/models/database.clj#L354-L357